JavaScript/JavaScript.sublime-syntax

```diff
@@ -207,13 +207,13 @@ contexts:
         8: storage.type.function.arrow.js
     - match: |-
         (?x)
-        (\b_?[A-Z][$\w]*)?
+        (\b_?[a-zA-Z_?.$]*)?
```

Comment:
Ah weird, I was the one who added that weird question mark thing, but I am pretty sure I copied it from an older syntax. Anyway I will update both of them.

As for the performance problems, do you think they were from my changes? I looked over them again and didn't see anything crazy unless it is something to do with negative/positive lookahead/lookbehinds making things slower.  Like changing

```diff
-    - match: \.(shift|sort|splice|unshift|pop|push|reverse)\b
 +    - match: (?<=\.)(shift|sort|splice|unshift|pop|push|reverse)\b(?=\()
```